components/widgets/menu/menu.jsx

```diff
@@ -21,6 +21,46 @@ export default class Menu extends React.PureComponent {
         this.node = React.createRef();
     }
 
+    hideUnneededDividers = () => {
+        if (this.node.current === null) {
+            return;
+        }
+
+        const children = Object.values(this.node.current.children).slice(0, this.node.current.children.length);
+
+        // Hiding dividers at beginning and duplicated ones
+        let prevWasDivider = false;
+        let isAtBeginning = true;
+        for (const child of children) {
+            if (child.classList.contains('menu-divider') || child.classList.contains('mobile-menu-divider')) {
+                if (isAtBeginning || prevWasDivider) {
+                    child.style.display = 'none';
```

Comment:
Why hide? Can't it be filtered out on the children itself before passing into Menu component?

Reply:
Not really, because it depends on the rendering of the children, so... we need to have everything rendered before we can figure out which ones I have to hide.